pyproject.toml

```
@@ -25,32 +26,33 @@ classifiers = [
     "Operating System :: Unix",
     "Operating System :: MacOS",
 ]
-requires-python = ">=3.9"
-dependencies = [
-    "aws-cdk-lib==2.48.0",
-    "aws-cdk.aws-lambda-python-alpha",
-    "constructs>=10.0.0,<11.0.0",
-    "boto3",
-    "opensearch-py",
-    "requests",
-    "python-jose"
-]
 
-[project.optional-dependencies]
-dev = [
-    "pytest==6.2.5",
-    "moto"
-]
+[tool.poetry.dependencies]
+python = ">=3.9,<4"
+# Required for aws-cdk.aws-lambda-python-alpha
+aws-cdk-lib = ">=2.66.0,<3.0.0"
+#aws-cdk-lib = "2.48.0"
+boto3 = "^1.26.78"
+"aws-cdk.aws-lambda-python-alpha" = "^2.66.0a0"
+constructs = ">=10.0.0,<11.0.0"
+black = "^23.1.0"
+
+[tool.poetry.group.tests.dependencies]
+# Installed by default, can be installed without these dependencies using `poetry install --without tests`
+pytest = "==6.2.5"
+moto = "^4.1.3"
+hypothesis = "^6.68.2"
+
+[tool.poetry.group.cdk]
```

```suggestion
[tool.poetry.group.lambda_code]
```
I think this is meant to be if we want to install the dependencies for the lambda code as well? I was a bit confused by the name, and I'm not sure mine is a great suggestion either, so feel free to bikeshed that. But cdk made me think this was installing the aws cdk or something similar to that.

---

Yeah, I didn't love the name either. Maybe `lambda_dev`?